server/src/__tests__/completions.test.ts

```diff
@@ -870,4 +870,59 @@ describe('On Completion', () => {
       resultOnStringContent.find((item) => item.label === 'deltask')
     ).toBeUndefined()
   })
+
+  it('provides common directories completion items where it is appropriate', async () => {
```

Comment:
During my test of this branch, diagnostics are fixed, the variables are defined, but I don't get completion or hover on ${base_bindir} and others.

Reply:
I don't think it is that urgent., I had tried on bash functions, so I'll look into python now ;)